cpp/src/io/json/json_tree.cu

```diff
@@ -487,13 +487,59 @@ std::pair<rmm::device_uvector<size_type>, rmm::device_uvector<size_type>> hash_n
   device_span<TreeDepthT const> node_levels,
   device_span<size_type const> node_type,
   device_span<NodeIndexT const> parent_node_ids,
+  bool is_array_of_arrays,
+  bool is_enabled_lines,
+  tree_meta_t const& d_tree,
   rmm::cuda_stream_view stream,
   rmm::mr::device_memory_resource* mr)
 {
   CUDF_FUNC_RANGE();
   auto const num_nodes = parent_node_ids.size();
   rmm::device_uvector<size_type> col_id(num_nodes, stream, mr);
 
+  // if level 1 is a list, then it should take its list index as key!
+  // (level 1 for values, level 0 for values-JSONLines)
+  // copy nodes of level 1 children. (level 2) and their parent_node_id
+  // exclusive scan by key -> get their indices and scatter to their node id
+  // use this index for hashing at level 2
```

Comment:
I added an example in the comment explaining the need for list_indices.
BFS may not apply here. It may be considered DFS. Each node computes its hash by walking from its node upto the root.

(2) Yes. Added example might clarify this.

Reply:
Thanks for clarifying. Agreed, I referred to BFS just as a means to describe how we identify the "data nodes". Not saying we would be using BFS in the algorithm.